redwoods-insurance-mobile/redwoods-insurance/ObjectLayoutDataSource.swift

```diff
@@ -91,83 +89,46 @@ class ObjectLayoutDataSource: NSObject {
 		SalesforceLogger.e(type(of: self), message: "Failed to successfully complete the REST request. \(errorDescription)")
 	}
 
-	/// Retrieves the compact layout metadata for the given object type, and
-	/// based on the field list, constructs a request to retrieve the fields
-	/// for the object with the given ID.
+	/// Retrieves the compact layout for the `objectType` and then fetches the
+	/// fields specified in the layout for the object identified by `objectId`.
 	///
-	/// - Parameters:
-	///   - objectType: The type of object to request from the server.
-	///   - objectId: The unique identifier of the object to request from the server.
-	///   - completionHandler: The closure to call when the request completes.
-	///   - request: The request that was built, which can be used to retrieve
-	///     the fields for the object.
-	private func buildRequestFromCompactLayout(forObjectType objectType: String, objectId: String, completionHandler: @escaping (_ request: RestRequest) -> Void) {
-		let layoutRequest = RestRequest(method: .GET, path: "/v44.0/compactLayouts?q=\(objectType)", queryParams: nil)
-		layoutRequest.parseResponse = false
-		RestClient.shared.send(request: layoutRequest, onFailure: handleError) { response, _ in
-			guard let responseData = response as? Data else { return }
-			do {
-				// The root object of the compact layouts response is a dictionary
-				// that maps object type names to their layouts. The first step
-				// is to find the requested object type in the dictionary.
-				let decodedJSON = try JSONDecoder().decode([String: CompactLayout].self, from: responseData)
-				guard let layout = decodedJSON[objectType] else {
-					SalesforceLogger.e(type(of: self), message: "Missing \(objectType) object type in response.")
-					return
-				}
+	/// When it successfully completes, the `fields` property is set, and the
+	/// delegate is notified of the update.
+	@objc func fetchData() {
+		guard !self.objectId.isEmpty else { return }
+		let queryParams: [String: Any] = ["layoutTypes": "Compact"]
+		let layoutRequest = RestRequest(method: .GET, path: "/v46.0/ui-api/record-ui/\(self.objectId)",
```

Comment:
Can we make that please `47.0`? ;-)

Reply:
This was refactored to utilize the static let constant APIVERSION